core/src/smartcontracts/isi/domain.rs

```diff
@@ -158,6 +165,10 @@ pub mod isi {
                 })?;
             }
 
+            if !wsv.remove_asset_register_tx(&asset_definition_id) {
```

Comment:
Maybe we should store this info in `AssetDefinition` directly? Not it wsv?

Reply:
that would make more sense

Reply:
This was a [suggestion](https://github.com/hyperledger/iroha/pull/3310#discussion_r1142115067) by @appetrosyan to save space. Though if instead of a map we're storing just a single hash with every `AssetDefinition`, that might make the logic even more lightweight.

Reply:
yeah, we don't want to store redundant info on chain so storing it in `wsv` is the way to go

Reply:
IMO, you should consider storing a hash of the block that registered the transaction, not hash of a transaction. Then when queried you should search the block for the actual transaction